x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/rules_stats_collector.ts

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { Logger } from '@kbn/core/server';
+import type { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import type { CspmRulesStats } from './types';
+import { LATEST_FINDINGS_INDEX_DEFAULT_NS } from '../../../../common/constants';
+
+interface DocCount {
+  doc_count: number;
+}
+
+interface BenchmarkName {
+  metrics: { 'rule.benchmark.name': string };
+}
+
+interface BenchmarkId {
+  metrics: { 'rule.benchmark.id': string };
+}
+
+interface BenchmarkVersion {
+  metrics: { 'rule.benchmark.version': string };
+}
+interface RuleName {
+  metrics: { 'rule.name': string };
+}
+interface RuleSection {
+  metrics: { 'rule.section': string };
+}
+interface RuleVersion {
+  metrics: { 'rule.version': string };
+}
+interface RuleNumber {
+  metrics: { 'rule.benchmark.rule_number': string };
+}
+interface PostureType {
+  metrics: { 'rule.benchmark.posture_type': string };
+}
+
+interface RulesStats {
+  rules: {
+    buckets: RuleEntity[];
+  };
+}
+
+interface RuleEntity {
+  key: string; // rule_id
+  rule_name: { top: RuleName[] };
+  rule_section: { top: RuleSection[] };
+  rule_version: { top: RuleVersion[] };
+  rule_number: { top: RuleNumber[] };
+  posture_type: { top: PostureType[] };
+  benchmark_id: { top: BenchmarkId[] };
+  benchmark_name: { top: BenchmarkName[] };
```

Comment:
i think we should add `benchmark.version` too
https://github.com/elastic/kibana/blob/36cce21ad7bf1a70531c9831008c306dcd5fb552/x-pack/plugins/cloud_security_posture/common/schemas/csp_rule_template_metadata.ts#L40